src/Counter.tsx

```diff
@@ -0,0 +1,70 @@
+import React, { Props } from 'react';
+import { Button } from './Button';
+import './Counter.scss';
+
+export class Counter extends React.Component<Props<State>, State> {
+  state: State = {
+    count: 0,
+  };
+
+  addOne = () => {
+    this.state.count += 1;
+
+    this.setState((prevState: State) => {
+      return {
+        count: prevState.count,
+      };
+    });
+  };
+
+  add100 = () => {
+    this.state.count += 100;
+
+    this.setState((prevState: State) => {
+      return {
+        count: prevState.count,
+      };
+    });
+  };
+
+  increase = () => {
+    this.addOne();
+
+    if ((this.state.count - 1) % 5 === 0) {
+      this.add100();
+    }
+
+    this.setState((prevState: State) => {
+      return {
+        count: prevState.count,
+      };
+    });
+  };
+
+  render() {
+    const { count } = this.state;
+    const { addOne, add100, increase } = this;
+
+    return (
+      <div className="Counter">
+        <h1 className="Counter__title">
+          {`Count: ${count}`}
+        </h1>
+        <div className="Counter__buttons">
+          <Button
+            action={addOne}
```

Comment:
For me it's more understandable to use 'this' here instead of did destructuring if we use class functions (I didn't see such code before, so I don't know is it better or not):
```suggestion
            action={this.addOne}
```